cognite/neat/graph/stores/_base.py

```diff
@@ -1,291 +1,165 @@
+        rules: InformationRules | None = None,
     ):
         _start = datetime.now(pytz.utc)
-        self.graph = graph or Graph()
-        self.base_prefix: str = base_prefix
-        self.namespace: Namespace = namespace
-        self.prefixes: dict[str, Namespace] = prefixes
-
-        self.rdf_store_query_url: str | None = None
-        self.rdf_store_update_url: str | None = None
-        self.returnFormat: str | None = None
-        self.df_cache: pd.DataFrame | None = None
-        self.internal_storage_dir: Path | None = None
-        self.graph_name: str | None = None
-        self.internal_storage_dir_orig: Path | None = None
-        self.storage_dirs_to_delete: list[Path] = []
-        self.queries = _Queries(self)
+        self.graph = graph
         self.provenance = Provenance(
             [
                 Change.record(
                     activity=f"{type(self).__name__}.__init__",
                     start=_start,
                     end=datetime.now(pytz.utc),
-                    description="Initialize graph store",
+                    description=f"Initialize graph store as {type(self.graph.store).__name__}",
                 )
             ]
         )
+        self.rules = rules
 
-    @abstractmethod
-    def _set_graph(self) -> None:
-        raise NotImplementedError()
-
-    def init_graph(
-        self,
-        rdf_store_query_url: str | None = None,
-        rdf_store_update_url: str | None = None,
-        graph_name: str | None = None,
-        base_prefix: str | None = None,
-        returnFormat: str = "csv",
-        internal_storage_dir: Path | None = None,
-    ):
-        """Initializes the graph.
-
-        Args:
-            rdf_store_query_url : URL towards which SPARQL query is executed, by default None
-            rdf_store_update_url : URL towards which SPARQL update is executed, by default None
-            graph_name : Name of graph, by default None
-            base_prefix : Base prefix for graph namespace to change if needed, by default None
-            returnFormat : Transport format of graph data between, by default "csv"
-            internal_storage_dir : Path to directory where internal storage is located,
-                                   by default None (in-memory storage).
-
-        !!! note "internal_storage_dir"
-            Used only for Oxigraph
-        """
-        logging.info("Initializing NeatGraphStore")
-        self.rdf_store_query_url = rdf_store_query_url
-        self.rdf_store_update_url = rdf_store_update_url
-        self.graph_name = graph_name
-        self.returnFormat = returnFormat
-        self.internal_storage_dir = Path(internal_storage_dir) if internal_storage_dir else None
-        self.internal_storage_dir_orig = (
-            self.internal_storage_dir if self.internal_storage_dir_orig is None else self.internal_storage_dir_orig
-        )
+        if self.rules:
+            self._upsert_prefixes(self.provenance, self.graph, self.rules)
```

Comment:
If you do the comment below, then this becomes 
```suggestion
            self._upsert_prefixes(self.rules.prefixes.values())
```
which is more readable to me